front50/front50-core/src/test/java/com/netflix/spinnaker/front50/echo/EchoServiceTest.java

```diff
@@ -0,0 +1,148 @@
+import com.github.tomakehurst.wiremock.junit5.WireMockExtension;
+import com.netflix.spectator.api.NoopRegistry;
+import com.netflix.spinnaker.config.DefaultServiceClientProvider;
+import com.netflix.spinnaker.config.OkHttpMetricsInterceptorProperties;
+import com.netflix.spinnaker.config.RetrofitConfiguration;
+import com.netflix.spinnaker.config.okhttp3.DefaultOkHttpClientBuilderProvider;
+import com.netflix.spinnaker.config.okhttp3.RawOkHttpClientFactory;
+import com.netflix.spinnaker.front50.config.EchoConfiguration;
+import com.netflix.spinnaker.front50.config.StorageServiceConfigurationProperties;
+import com.netflix.spinnaker.front50.model.plugins.PluginEvent;
+import com.netflix.spinnaker.front50.model.plugins.PluginEventType;
+import com.netflix.spinnaker.front50.model.plugins.PluginInfo;
+import com.netflix.spinnaker.kork.dynamicconfig.DynamicConfigService;
+import com.netflix.spinnaker.kork.retrofit.Retrofit2ServiceFactoryAutoConfiguration;
+import com.netflix.spinnaker.kork.retrofit.RetrofitServiceFactoryAutoConfiguration;
+import com.netflix.spinnaker.okhttp.OkHttp3MetricsInterceptor;
+import com.netflix.spinnaker.okhttp.OkHttpClientConfigurationProperties;
+import com.netflix.spinnaker.okhttp.Retrofit2EncodeCorrectionInterceptor;
+import com.netflix.spinnaker.okhttp.SpinnakerRequestHeaderInterceptor;
+import com.netflix.spinnaker.okhttp.SpinnakerRequestInterceptor;
+import com.netflix.spinnaker.retrofit.Retrofit2ConfigurationProperties;
+import com.netflix.spinnaker.retrofit.RetrofitConfigurationProperties;
+import java.util.List;
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.test.context.TestPropertySource;
+import retrofit.RequestInterceptor;
+
+@SpringBootTest(
+    classes = {
+      EchoServiceTest.EchoServiceTestConfig.class,
+      EchoConfiguration.class,
+      DefaultServiceClientProvider.class,
+      DynamicConfigService.NoopDynamicConfig.class,
+      StorageServiceConfigurationProperties.class,
+      NoopRegistry.class,
+      RetrofitServiceFactoryAutoConfiguration.class,
+      Retrofit2ServiceFactoryAutoConfiguration.class,
+      RetrofitConfiguration.class,
+      Retrofit2ConfigurationProperties.class,
+      DefaultOkHttpClientBuilderProvider.class,
+      OkHttp3MetricsInterceptor.class,
+      RawOkHttpClientFactory.class,
+      RetrofitConfigurationProperties.class,
+      OkHttpClientConfigurationProperties.class,
+      OkHttpMetricsInterceptorProperties.class,
+      ObjectMapper.class
+    })
+@TestPropertySource(
+    properties = {"services.echo.enabled=true", "services.echo.base-url=https://localhost:56789"})
```

Comment:
hard-coded ports give me the willies.  Can you switch to a dynamic port, likely with `@DynamicPropertySource` ?

Reply:
Actually this was the test before fix where the test fails before even the call is made. And I made necessary arrangements to use dynamic port post fix.

Reply:
I made changes to EchoConfiguration for better testability. Dynamic port is not possible like the way it does post fix.

Reply:
I need to change the EchoConfiguration first and add test to demonstrate the issue. Shall I do that?